File: 0392-is-subsequence/0392-is-subsequence.js
var isSubsequence = function (s, t) {
    if(!s.length) return true;

    let chrIdx = 0;

    for (let i = 0; i < t.length; i++) {
        if (s[chrIdx] === t[i]) {
            chrIdx++;
            if (chrIdx === s.length) return true
        }
    }

    return false;
};

File: 0392-is-subsequence/0392-is-subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        if(s.Length == 0) return true;
        int Idx = 0;
        for (int i = 0; i < t.Length; i++)
        {
            if(s[Idx] == t[i]){
                Idx++;
                if(s.Length == Idx) return true;
            }
        }
        return false;
    }
}